lib/release/api.ts

```diff
@@ -104,6 +105,7 @@ export async function create(req: Request) {
 
 	const res = { release, serviceImages: {} } as Response;
 
+	const imgDescriptors = parse(req.composition);
```

Comment:
The approach you went for presents to `release` a composition that is no longer canonical because it may contain `contract` keys. I think a better approach is to pass a dictionary of contracts per service as a second optional argument. It would also save you from having to re-parse here.

Reply:
I suspected you would. Let me look into it.